### ISSUE TITLE
users.getById({id}) doesn't work

### ISSUE THREAD

**Issue by AlexeyBond** [octokit/octokit.js#717]
```
> github.users.getById({id:'alexeybond'}).catch(console.log)
Promise {
  <pending>,
  domain: 
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
> { [Error: {"message":"Not Found","documentation_url":"https://developer.github.com/v3/users/#get-a-single-user"}]
  message: '{"message":"Not Found","documentation_url":"https://developer.github.com/v3/users/#get-a-single-user"}',
  code: 404,
  status: 'Not Found',
  headers: 
   { ... }
}
```
But this works:
```
> github.request({url:'/users/alexeybond'}).then(console.log)
Promise {
  <pending>,
  domain: 
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
> { meta: 
   { ... },
  data: 
   { login: 'AlexeyBond',
...
 } }
```
[Documentation](https://octokit.github.io/rest.js/#api-Users-getById) looks like `getById` calls `/user/:id` instead of `/users/:id`.

**Comment by synicalsyntax** [comment]
@AlexeyBond You might be looking for the [users.getForUser()](https://octokit.github.io/rest.js/#api-Users-getForUser) method instead? That method corresponds to [this part of the API](https://developer.github.com/v3/users/#get-a-single-user).

`users.getByID()` is meant to retrieve users not by username, but by a numerical ID, such as `7383049` (for you). It corresponds to the endpoint at `https://api.github.com/user/7383049` instead of `https://api.github.com/users/AlexeyBond`.

**Comment by gr2m** [comment]
Hey Alexey 👋 Cynthia is right, I’m sorry for the confusion!

If you like, you could send a pull request to make the API docs more clear? I think you are not the first who got confused by it, if I recall correctly I run into it myself in the past :)

You can amend the method description and add a description for the `id` parameter here:

https://github.com/octokit/rest.js/blob/34109d614958f8e423beb94c6959920e79c14862/lib/routes.json#L9677-L9687

Please let me know if you have any questions

**Comment by AlexeyBond** [comment]
@synicalsyntax Thanks, that was the method I was looking for.

@gr2m I have a question: was there any special reason for setting type of `id` parameters of methods `users.getById` and `repos.getById` to `"string"` instead of `"number"` and will it break anything if I change it to `"number"`?

**Comment by gr2m** [comment]
That is a good question, I don’t know why it’s `"string"`, as far as I know all the IDs used in the REST API are numeric. I’ve taken over the maintenance of this project a few months ago, these settings have been there long before I joined the project. I’d leave it for now, I’ll do an audit of the whole `routes.json` file soon any as it’s not up to date.

For example, I don't know why there is a `getById` anyway, as it is not documented in the [REST API](https://developer.github.com/v3/). My guess is that it was at some point but then removed it, as it’s considered an internal route.

So in conclusion, I’d leave the method and `id` parameter as is to avoid breaking anything, but we can certainly improve the descriptions for more clarity

**Comment by gr2m** [comment]
Documentation got updated: https://octokit.github.io/rest.js/#api-Users-getById
Thanks to both of you :)
